src/tests/categoryController.spec.ts

```diff
@@ -0,0 +1,317 @@
+		};
+		const req = mockReq({
+			body: category,
+			params: { id: 0 },
+		});
+		const res = mockRes({
+			status: function(s: any) {
+				this.statusCode = s;
+				return this;
+			},
+		});
+		const getByIdStub = sinon.stub(CategoryService, 'getById').resolves(mockCategories[req.params.id]);
+		const saveStub = sinon.stub(CategoryService, 'save').resolvesArg(0);
+		await CategoryController.editCategory(req, res);
+		expect(res.statusCode).to.equal(204);
+	});
+	it('should fail to edit category because name is already in use, return 409', async () => {
+		const category = {
+			name: 1,
+			description: 123,
+			services: 2,
+		};
+		const req = mockReq({
+			body: category,
+			params: { id: 0 },
+		});
+		const res = mockRes({
+			status: function(s: any) {
+				this.statusCode = s;
+				return this;
+			},
+		});
+		const getByIdStub = sinon.stub(CategoryService, 'getById').resolves(mockCategories[req.params.id]);
+		const saveStub = sinon.stub(CategoryService, 'save').throws();
+		await CategoryController.editCategory(req, res);
+		expect(res.statusCode).to.equal(409);
+	});
+
+	it('should delete the specified category, return 204', async () => {
+		const req = mockReq({
+			params: { id: 0 },
+		});
+		const res = mockRes({
+			status: function(s: any) {
+				this.statusCode = s;
+				return this;
+			},
+		});
+		const getByIdStub = sinon.stub(CategoryService, 'getById').resolves(mockCategories[req.params.id]);
+		const deleteByIdStub = sinon.stub(CategoryService, 'deleteById').resolves(new DeleteResult());
+		await CategoryController.deleteCategory(req, res);
+		expect(res.statusCode).to.equal(204);
+	});
+	it('should fail to delete the specified category because it does not exist, return 404', async () => {
+		const req = mockReq({
+			params: { id: 0 },
+		});
+		const res = mockRes({
+			status: function(s: any) {
+				this.statusCode = s;
+				return this;
+			},
+		});
+		const getByIdStub = sinon.stub(CategoryService, 'getById').throws();
+		await CategoryController.deleteCategory(req, res);
+		expect(res.statusCode).to.equal(404);
+	});
+
+	it('should return error when getoneId service throws error', async () => {
+		const req = mockReq({
+			params: { id: 0 },
+		});
+		const res = mockRes({
+			status: function() {
+				return {
+					send: (e: any) => e,
+				};
+			},
+		});
+		const getByIdStub = sinon.stub(CategoryService, 'getById').throws('exception');
+		const result = await CategoryController.getOneById(req, res);
+		expect(result).equals('Category not found');
+	});
+	it('should return error when validation of new category throws error', async () => {
+		const req = mockReq({
+			params: { id: 0 },
+		});
+		const res = mockRes({
+			status: function() {
+				return {
+					send: (e: any) => e,
+				};
+			},
+		});
+		const validateStub = sinon.stub(validator, 'validate').resolves([new ValidationError()]);
+		const result = await CategoryController.newCategory(req, res);
+		expect(result).to.not.be.null;
```

Comment:
Den skal tjekke length eller not empty i stedet, da der ved ingen validation errors kommer et tomt array tilbage, som heller ikke er null